rag_experiment_accelerator/config/index_config.py

```diff
@@ -1,59 +1,78 @@
+        chunking (ChunkingConfig):
+            Configuration for chunking documents.
+        embedding_model (EmbeddingModelConfig):
+            Configuration for the embedding model.
+        sampling (SamplingConfig):
+            Configuration for sampling documents.
     """
 
-    index_name_prefix: str
-    preprocess: bool
-    chunk_size: int
-    overlap_size: int
-    embedding_model: EmbeddingModel
-    ef_construction: int
-    ef_search: int
-    sampling_percentage: int = 0
-    generate_title: bool = False
-    generate_summary: bool = False
-    override_content_with_summary: bool = False
+    index_name_prefix: str = "idx"
+    ef_construction: int = 400
+    ef_search: int = 400
+    chunking: ChunkingConfig = field(default_factory=ChunkingConfig)
+    embedding_model: EmbeddingModelConfig = field(default_factory=EmbeddingModelConfig)
+    sampling: SamplingConfig = field(default_factory=SamplingConfig)
+
+    def label_properties(self) -> dict:
+        """
+        Returns properties subset used for labeling.
+        """
+        properties = {
+            "idx": self.index_name_prefix,
+            "efc": self.ef_construction,
+            "efs": self.ef_search,
+            "emn": self.embedding_model.model_name.lower(),
+            "sp": self.sampling.percentage,
+        }
+
+        properties.update(self.chunking.label_properties())
+
+        return properties
+
+    @classmethod
+    def from_label_properties(cls, properties: dict) -> "IndexConfig":
+        """
+        Creates IndexConfig from the dictionary with properties.
+        Reverse of label_properties().
+        """
+
+        return IndexConfig(
+            index_name_prefix=properties["idx"],
+            ef_construction=int(properties["efc"]),
+            ef_search=int(properties["efs"]),
+            chunking=ChunkingConfig.from_label_properties(properties),
+            embedding_model=EmbeddingModelConfig(model_name=properties["emn"]),
+            sampling=SamplingConfig(percentage=properties["sp"]),
+        )
 
     def index_name(self) -> str:
         """
         Returns index name from the fields.
         Reverse of IndexConfig.from_index_name().
         """
-        index_name = (
-            f"{self.index_name_prefix}"
-            f"_p-{int(self.preprocess)}"
-            f"_cs-{str(self.chunk_size)}"
-            f"_o-{str(self.overlap_size)}"
-            f"_efc-{str(self.ef_construction)}"
-            f"_efs-{str(self.ef_search)}"
-            f"_sp-{str(self.sampling_percentage)}"
-            f"_t-{int(self.generate_title)}"
-            f"_s-{int(self.generate_summary)}"
-            f"_oc-{int(self.override_content_with_summary)}"
-            f"_{str(self.embedding_model.name.lower())}"
+        index_name = "_".join(
+            [f"{key}-{value}" for (key, value) in self.label_properties().items()]
         )
         if index_name.startswith("_") or index_name.startswith("-"):
```

Comment:
If index_name is longer than 128 chars error will be thrown. It should be reduced somehow in that case